riff-raff/app/assets/javascripts/form-autocomplete.coffee

```diff
@@ -85,6 +94,13 @@ $ ->
       source:serverUrl
       minLength:0
 
+    addFavouriteProjectButton = $('#add-favourite-project-button')
+    updateFavouriteButton = ->
+      console.log("fave change")
+      addFavouriteProjectButton.prop('disabled', /^\s*$/.test(input.val()))
+    input.on('change', updateFavouriteButton)
+    input.on('keyup', updateFavouriteButton)
```

Comment:
What's this doing? My assumption is it's removing the `disabled` attribute?

Reply:
Yep that's correct. We're listening on the keyup event and calling the `updateFavouriteButton` callback when it happens (the callback is defined just above).

Reply:
To clarify further: in the callback we set the `disabled` prop of the button, depending on whether the project input is empty or not.

Reply:
👍🏽  I'm a little confused by the regex - could we use [`removeAttribute`](https://developer.mozilla.org/en-US/docs/Web/API/Element/removeAttribute)? Or is this because this is coffeescript?

Reply:
> To clarify further: in the callback we set the `disabled` prop of the button, depending on whether the project input is empty or not.

Ah! Would something like this be clearer?

```js
addFavouriteProjectButton.disabled = input.val() === ""
```

Reply:
Ah yep we could definitely do that! No this isn't coffeescript specific I don't think, it's using [`prop` from jQuery](https://api.jquery.com/prop/#prop2) - old school! I quite like this approach as it doesn't require two branches (one for removing and one for adding), we take this approach elsewhere in the codebase too. Would this make it clearer do you think?

```suggestion
      projectInputIsEmpty = /^\s*$/.test(input.val())
      addFavouriteProjectButton.prop('disabled', projectInputIsEmpty)
    input.on('change', updateFavouriteButton)
    input.on('keyup', updateFavouriteButton)
```

Reply:
> addFavouriteProjectButton.disabled = input.val() === ""

Oops our last messages overlapped. Yep I like that approach 👍  The regex was a little extra check which meant you couldn't hit space a bunch of times and favourite _that_ project name, but maybe not worth the additional complexity added?

Reply:
> The regex was a little extra check which meant you couldn't hit space a bunch of times and favourite that project name

`input.val().trim()` would do the same thing and is clearer, at least to me, than regex in this instance.

Reply:
Ah yep that's much better, thanks for the suggestion.

Reply:
I've made this change. Setting `.disabled` directly doesn't seem to have the desired effect (I guess that is a CoffeeScript thing) so I've stuck with using `prop`.